### ISSUE TITLE
Specific Emacs Issue with Blink App connecting over ssh (not mosh)

### ISSUE THREAD

**Issue by prairie-guy** [mobile-shell/mosh#1099]
I love mosh and Blink on an IPad. Amazing work! Accordingly, I’ve been trying to get all my boxes to be Blink/mosh capable. Sadly, on two clients I cannot get mosh installed. (Can’t control firewall and am not root). I still want to use Blink over ssh. 

This is a very specific problem: When I use the Blink App to ssh a Ubuntu box, I cannot open any files with Emacs. (I have replicated this problem on multiple Ubuntu boxes with different default setup, I.e., not clones.)

- `emacs  .bashrc` only opens a `scratch` buffer
- Doesn’t matter if I change it to `emacs  ~/.bashrc` or `emacs  /home/me/.bashrc`
- I have `rm -fr.emacs.d` and started clean. Same problem
- `emacs —debug-init` shows no clues
- I have tried emacs versions 24-26.
- Python and Julia (and other programs) load files as expected.
- I have tried clean .bashrc and .bash_profile files. No help

**This only happens with the Blink App. I have no problems if I ssh from OS X Terminal, OS X iTerm2 or iPadOS using the App Remote.**

Am I missing a configuration issue or is there a reason that using emacs over ssh with other programs works, but not Blink?

Any suggestions or next steps in terms of debugging would be much appreciated.

I could use one of these other programs, but I’m hooked on Blink/mosh.

**Comment by cgull** [comment]
Do you have this issue with Mosh on a computer?  (Even ssh to the server, and then Mosh to itself.)  If it's a Blink-specific problem, I would suggest taking it up at https://github.com/blinksh/blink

That said, this sounds like a higher-level issue, perhaps your `TERM` variable isn't set to a value that Emacs likes?

Also, if you can describe more precisely what happens when you run Emacs, what you see, and what you expect to see, it'll be easier to debug problems.  "When I type `emacs` at the command line, I get the error message `the parrot is dead, pid 1319`" is more useful than "I cannot open any files".

**Comment by prairie-guy** [comment]
Thank you for responding. Your suggestion of investigating TERM was a good idea. Here is what I discovered:
```
TERM=Value		Environment	Application	OS		Command		Result
“term-256color”		shell		Bash		Ubuntu		env             --

“dumb”			emacs		Terminal	Ubuntu		mosh		OK
“dumb”			emacs		Terminal	Ubuntu		ssh		OK

“dumb”			emacs		Blink App	iOS		mosh		OK
“xterm-256color”	emacs		Blink App	iOS		ssh		Fails

“dumb”			emacs		Prompt App	iOS		ssh		OK
```
For some reason, only when using the Blink App over ssh (vs. mosh) is the emacs internal value `TERM=term-256color` When using mosh or other implementations of ssh, the emacs internal value is `TERM=dumb`

The only kluge fix I have come up with is to set `export TERM=screen` at the shell. (But it is ugly screen.)  I have tried, but have not been able to set emacs internally to `TERM=dumb` (`export TERM=screen` at the shell fails with an emacs error that `dumb` is not a powerful enough for emacs.)

I don’t know if this is a Blink App issue, a mosh issue or an emacs issue. I’m stuck and would appreciate any advice.

Thanks in advance
